src/v/storage/tests/utils/random_batch.cc

```diff
@@ -153,4 +154,19 @@ ss::circular_buffer<model::record_batch> make_random_batches(model::offset o) {
     return make_random_batches(o, get_int(2, 30));
 }
 
+model::record_batch_reader make_random_memory_record_batch_reader(
+  model::offset offset, int batch_size, int n_batches, bool allow_compression) {
+    return model::make_generating_record_batch_reader(
+      [offset, batch_size, n_batches, allow_compression]() mutable {
+          if (n_batches-- <= 0) {
```

Comment:
That requires some thought!

This might be easier, at least in my head:
```cpp
if (n_batches--) {
    do_the_thing();
}
dont_do_the_thing();
```

But it's a very minor nit.

Reply:
implemented